packages/hub/src/components/NavigationBlocker/InterceptedLinkModal.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRouter } from "next/navigation";
```

Comment:
I think there should be a small README.md in the NavigationBlocker/ directory. Right now, it's not very clear what the structure is of this / where to begin reading it. A simple explanation, with maybe a simple mermaid file, would help.

Reply:
Here's a quick first pass at the diagram, by Claude.

<img width="966" alt="image" src="https://github.com/user-attachments/assets/7a477579-5045-4fc3-abba-2a79804aee5a">

Reply:
```mermaid
graph TD
    subgraph NavigationBlocker[Navigation Blocker Module]
        A[NavigationBlockerProvider]:::blockerNode
        B[useBlockNavigation]:::blockerNode
        C[InterceptedLinkModal]:::blockerNode
        D[Navigation Blocker Context]:::blockerNode
        E[Custom Link Component]:::blockerNode
    end

    F[User Components]:::externalNode
    G[Next.js Router]:::externalNode

    A -->|Provides context| B
    A -->|Renders| C
    B -->|Registers blocker| D
    E -->|Checks| D
    E -->|Triggers| C
    F -->|Use| B
    F -->|Use| E
    E -->|Intercepts| G

    classDef blockerNode fill:#e6f3ff,stroke:#0056b3,stroke-width:2px,color:#003366;
    classDef externalNode fill:#fff5e6,stroke:#ff9900,stroke-width:2px,color:#804d00;
    linkStyle default stroke:#666,stroke-width:2px;
 ```

Reply:
As I said during our call, I feel like README files are hard to maintain (e.g. when you reorganize the component or rename its functions) and often end up write-only because it's easy to forget that the README is there when you're looking at the code.

I added a comment in `ExitConfirmationWrapper/index.ts` that should clarify things.